internal/controller/controller.go

```diff
@@ -79,10 +84,29 @@ func (c *httpController) AddRoutes() error {
 }
 
 // StartRecording starts a recording session based on the values in the request.
-// An error is returned if the request data is incomplete or a record or replay session is currently running.
+// An error is returned if the request data is incomplete.
 func (c *httpController) startRecording(writer http.ResponseWriter, request *http.Request) {
-	//TODO implement me using TDD
-	writer.WriteHeader(http.StatusNotImplemented)
+	startRequest := dtos.RecordRequest{}
+
+	if err := json.NewDecoder(request.Body).Decode(&startRequest); err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		_, _ = writer.Write([]byte(fmt.Sprintf("%s: %v", failedRequestJSON, err)))
+		return
+	}
+
+	if startRequest.Duration == 0 && startRequest.EventLimit == 0 {
+		writer.WriteHeader(http.StatusBadRequest)
+		_, _ = writer.Write([]byte(failedRecordRequestValidate))
+		return
```

Comment:
should we also validate the value must be non-negative .. so some meaningful value to have?

Reply:
Good catch. Will add

Reply:
added